### PR TITLE
Remove account from identity and context; make account optional on hosts

### PR DESCRIPTION
**Pull request by kruai**
Addresses [ESSNTL-2530](https://issues.redhat.com/browse/ESSNTL-2530) & [ESSNTL-2531](https://issues.redhat.com/browse/ESSNTL-2531).

- Removes `account` from Identity
- Removes `account_number` from context
- Rejects requests whose Identity is missing org_id instead of account
- Makes `org_id` a required field on Host
- Makes `account` an optional field on Host
- Adds `org_id` to 3 snapshots
- Updates README with new base64 Identity and corresponding plaintext value

### CODE REVIEW

**Review by thearifismail — COMMENTED** [review]
Overall the PR LGTM.  Here are few observations to consider:

1. Same identity should be used in docs and code.  This may need updating `src/middleware/identity/utils.ts`.
2. Same `identity` should be used in `xjoin` and `host-inventory` code.
3. Though `org_id` will be a required parameter, searches using `account` should be available.  This is what Barnaby Court wanted to make sure that customer may still want to `account` instead of `org_id`.
4. The playgraound home page does not load.  The playground page, http://localhost:4000/graphql, shows:
```
Error: BadRequest
    at identity (/Users/aarif/Documents/dev-ws/insights/asa_xjoin/src/middleware/identity/impl.ts:21:25)
    at Layer.handle [as handle_request] (/Users/aarif/Documents/dev-ws/insights/asa_xjoin/node_modules/express/lib/router/layer.js:95:5)
    at trim_prefix (/Users/aarif/Documents/dev-ws/insights/asa_xjoin/node_modules/express/lib/router/index.js:323:13)
    at /Users/aarif/Documents/dev-ws/insights/asa_xjoin/node_modules/express/lib/router/index.js:284:7
    at Function.process_params (/Users/aarif/Documents/dev-ws/insights/asa_xjoin/node_modules/express/lib/router/index.js:341:12)
    at next (/Users/aarif/Documents/dev-ws/insights/asa_xjoin/node_modules/express/lib/router/index.js:275:10)
    at identityFallback (/Users/aarif/Documents/dev-ws/insights/asa_xjoin/src/middleware/identity/fallback.ts:10:5)
    at Layer.handle [as handle_request] (/Users/aarif/Documents/dev-ws/insights/asa_xjoin/node_modules/express/lib/router/layer.js:95:5)
    at trim_prefix (/Users/aarif/Documents/dev-ws/insights/asa_xjoin/node_modules/express/lib/router/index.js:323:13)
    at /Users/aarif/Documents/dev-ws/insights/asa_xjoin/node_modules/express/lib/router/index.js:284:7
    at Function.process_params (/Users/aarif/Documents/dev-ws/insights/asa_xjoin/node_modules/express/lib/router/index.js:341:12)
    at next (/Users/aarif/Documents/dev-ws/insights/asa_xjoin/node_modules/express/lib/router/index.js:275:10)
    at middleware (/Users/aarif/Documents/dev-ws/insights/asa_xjoin/node_modules/express-prom-bundle/src/index.js:204:5)
    at Layer.handle [as handle_request] (/Users/aarif/Documents/dev-ws/insights/asa_xjoin/node_modules/express/lib/router/layer.js:95:5)
    at trim_prefix (/Users/aarif/Documents/dev-ws/insights/asa_xjoin/node_modules/express/lib/router/index.js:323:13)
    at /Users/aarif/Documents/dev-ws/insights/asa_xjoin/node_modules/express/lib/router/index.js:284:7
```

Here is the console output:
```
$ npm run start

> xjoin-search@0.1.0 start
> tsc && LOG_PRETTY=true ts-node-dev --respawn src/app.ts

[INFO] 16:26:34 ts-node-dev ver. 1.1.8 (using ts-node ver. 9.1.1, typescript ver. 4.6.3)
[1657743997923] INFO (xjoin-search/12221 on MacBook-Pro): xjoin-search/development starting
    env: "development"
    type: "application"
[1657743997925] DEBUG (xjoin-search/12221 on MacBook-Pro): {
  "env": "development",
  "port": 4000,
  "logging": {
    "level": "trace",
    "pretty": true,
    "cloudwatch": {
      "enabled": false,
      "group": "xjoin-ci",
      "stream": "xjoin-ci",
      "intervalMs": 1000,
      "secret": "[Sensitive]",
      "region": "us-east-1"
    }
  },
  "es": {
    "nodes": "http://localhost:9200",
    "username": "",
    "password": "[Sensitive]"
  },
  "metrics": {
    "prefix": "xjoin_search_",
    "port": 4000,
    "path": "/metrics"
  },
  "queries": {
    "hosts": {
      "index": "test.hosts"
    },
    "maxBuckets": 10000
  }
}
    type: "application"
[1657743997959] DEBUG (xjoin-search/12221 on MacBook-Pro): connected to Elasticsearch
    type: "application"
[1657743997965] WARN (xjoin-search/12221 on MacBook-Pro): Identity fallback enabled, unsafe for production!
    type: "application"
[1657743998090] INFO (xjoin-search/12221 on MacBook-Pro): accepting connections
    url: "http://localhost:4000/graphql"
    type: "application"
[1657744050150] INFO (xjoin-search/12221 on MacBook-Pro): rejecting request for undefined "org_id"
    type: "application"
Error: BadRequest
    at identity (/Users/aarif/Documents/dev-ws/insights/asa_xjoin/src/middleware/identity/impl.ts:21:25)
    at Layer.handle [as handle_request] (/Users/aarif/Documents/dev-ws/insights/asa_xjoin/node_modules/express/lib/router/layer.js:95:5)
    at trim_prefix (/Users/aarif/Documents/dev-ws/insights/asa_xjoin/node_modules/express/lib/router/index.js:323:13)
    at /Users/aarif/Documents/dev-ws/insights/asa_xjoin/node_modules/express/lib/router/index.js:284:7
    at Function.process_params (/Users/aarif/Documents/dev-ws/insights/asa_xjoin/node_modules/express/lib/router/index.js:341:12)
    at next (/Users/aarif/Documents/dev-ws/insights/asa_xjoin/node_modules/express/lib/router/index.js:275:10)
    at identityFallback (/Users/aarif/Documents/dev-ws/insights/asa_xjoin/src/middleware/identity/fallback.ts:10:5)
    at Layer.handle [as handle_request] (/Users/aarif/Documents/dev-ws/insights/asa_xjoin/node_modules/express/lib/router/layer.js:95:5)
    at trim_prefix (/Users/aarif/Documents/dev-ws/insights/asa_xjoin/node_modules/express/lib/router/index.js:323:13)
    at /Users/aarif/Documents/dev-ws/insights/asa_xjoin/node_modules/express/lib/router/index.js:284:7
    at Function.process_params (/Users/aarif/Documents/dev-ws/insights/asa_xjoin/node_modules/express/lib/router/index.js:341:12)
    at next (/Users/aarif/Documents/dev-ws/insights/asa_xjoin/node_modules/express/lib/router/index.js:275:10)
    at middleware (/Users/aarif/Documents/dev-ws/insights/asa_xjoin/node_modules/express-prom-bundle/src/index.js:204:5)
    at Layer.handle [as handle_request] (/Users/aarif/Documents/dev-ws/insights/asa_xjoin/node_modules/express/lib/router/layer.js:95:5)
    at trim_prefix (/Users/aarif/Documents/dev-ws/insights/asa_xjoin/node_modules/express/lib/router/index.js:323:13)
    at /Users/aarif/Documents/dev-ws/insights/asa_xjoin/node_modules/express/lib/router/index.js:284:7
```

File: README.md [hunk 1]
```diff
@@ -98,14 +98,14 @@ Examples using curl:
...
---data-binary '{"query":"{hosts(limit:10,offset:0){meta{count,total}data{id account display_name}}}"}' \
+--data-binary '{"query":"{hosts(limit:10,offset:0){meta{count,total}data{id org_id display_name}}}"}' \
 http://localhost:4000/graphql
 
 # passing x-rh-identity object ( must be base64 encoded )
 curl \
 -H 'Content-Type: application/json' \
--H 'x-rh-identity: eyJpZGVudGl0eSI6eyJhY2NvdW50X251bWJlciI6InRlc3QiLCJ0eXBlIjoiVXNlciIsInVzZXIiOnsidXNlcm5hbWUiOiJ0dXNlckByZWRoYXQuY29tIiwiZW1haWwiOiJ0dXNlckByZWRoYXQuY29tIiwiZmlyc3RfbmFtZSI6InRlc3QiLCJsYXN0X25hbWUiOiJ1c2VyIiwiaXNfYWN0aXZlIjp0cnVlLCJpc19vcmdfYWRtaW4iOmZhbHNlLCJpc19pbnRlcm5hbCI6dHJ1ZSwibG9jYWxlIjoiZW5fVVMifX19' \
---data-binary '{"query":"{hosts(limit:10,offset:0){meta{count,total}data{id account display_name}}}"}' \
+-H 'x-rh-identity: eyJpZGVudGl0eSI6eyJvcmdfaWQiOiJ0ZXN0IiwidHlwZSI6IlVzZXIiLCJ1c2VyIjp7InVzZXJuYW1lIjoidHVzZXJAcmVkaGF0LmNvbSIsImVtYWlsIjoidHVzZXJAcmVkaGF0LmNvbSIsImZpcnN0X25hbWUiOiJ0ZXN0IiwibGFzdF9uYW1lIjoidXNlciIsImlzX2FjdGl2ZSI6dHJ1ZSwiaXNfb3JnX2FkbWluIjpmYWxzZSwiaXNfaW50ZXJuYWwiOnRydWUsImxvY2FsZSI6ImVuX1VTIn19fQ==' \
```
[the remark this answers is not in the document]
Comment (kruai) [reply README.md]: OK, I've added the missing `auth_type` field to the b64 identity in the README.

**Comment by thearifismail** [comment]
> @thearifismail For points 1 and 2, `auth_type` was already missing from those identities, but they're now updated to match.
> 
> For point 3... everyone is supposed to stop using `account`. If it turns out we do need to provide the ability to filter on that field, it will require separate development. We would need to add an `account` param to all of the HBI endpoints for which they're requesting this, and also add a new filter here in xjoin (since it can't be in the request context anymore).
> 
> And for point 4, the playground loads and accepts requests just fine for me. It looks like some request isn't including `org_id`. Did you do the usual steps from the README (`npm ci`, `npm run compile`, etc)?

Good for Point 1 and 2.
Point 3 to be discussed with Barnaby Court.
Point 4 appeared to have been caused by some cache corruption in my Chrome.  Tried Firefox and it worked as expected.

**Review by thearifismail — APPROVED** [review]
LGTM

**Comment by fijshion** [comment]
lgtm - just want to double check before merging - do all requests already include an org_id?

**Comment by kruai** [comment]
> lgtm - just want to double check before merging - do all requests already include an org_id?

Yep! I just double-checked with Keith, and he confirmed that all 3Scale requests are already sending it in both Stage and Prod.

**Pull request merged**